osu.Game/Graphics/UserInterface/BarGraph.cs

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using OpenTK;
+using OpenTK.Graphics;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Colour;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Sprites;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace osu.Game.Graphics.UserInterface
+{
+    public class BarGraph : FillFlowContainer<Bar>
+    {
+        private BarDirection direction = BarDirection.BottomToTop;
+        public new BarDirection Direction
+        {
+            get
+            {
+                return direction;
+            }
+            set
+            {
+                direction = value;
+                base.Direction = (direction & BarDirection.Horizontal) > 0 ? FillDirection.Vertical : FillDirection.Horizontal;
+                foreach (var bar in Children)
+                {
+                    bar.Size = (direction & BarDirection.Horizontal) > 0 ? new Vector2(1, 1.0f / Children.Count()) : new Vector2(1.0f / Children.Count(), 1);
+                    bar.Direction = direction;
+                }
+            }
+        }
+
+        public IEnumerable<float> Values
+        {
+            set
+            {
+                List<float> values = value.ToList();
+                List<Bar> graphBars = Children.ToList();
+                for (int i = 0; i < values.Count; i++)
+                    if (graphBars.Count > i)
+                    {
+                        graphBars[i].Length = values[i] / values.Max();
+                        graphBars[i].Size = (direction & BarDirection.Horizontal) > 0 ? new Vector2(1, 1.0f / values.Count) : new Vector2(1.0f / values.Count, 1);
+                    }
+                    else
+                        Add(new Bar
+                        {
+                            RelativeSizeAxes = Axes.Both,
+                            Size = (direction & BarDirection.Horizontal) > 0 ? new Vector2(1, 1.0f / values.Count) : new Vector2(1.0f / values.Count, 1),
+                            Length = values[i] / values.Max(),
+                            Direction = Direction,
+                            BackgroundColour = new Color4(0, 0, 0, 0),
+                        });
+            }
+        }
+    }
+
+    public class Bar : Container

```

Comment:
This is probably deserving of its own file given that it's being used in multiple places, along with BarDirection. I would namespace these inside Graphics.UserInterface.BarGraph.

I'm not sure about Bar having a background - given that >= 50% of the usage cases set it to transparent anyway and that it's static, it could be handled externally.